### PR TITLE
Created the widgets folder

### LINKED ISSUE #30: Dedup widgets
## Description
All common widgets across screens should be de-duplicated (dedup), copied and imported from lib/widgets/

List of widgets to dedup:
- EmailField
- PasswordField
- TextField (Rename NameField to this)

### PR DESCRIPTION
**Pull request by divy-arpit**
### Description

Widgets which were being repeated have been brought under a single folder

**Type of Change:** (Delete irrelevant options)
- Code

**Code/Quality Assurance Only** (Delete irrelevant options)

- Bug fix (non-breaking change which fixes an issue)

### How Has This Been Tested?

Tested on local machine

### Checklist:

- [X] My PR follows the style guidelines of this project
- [X] I have performed a self-review of my own code or materials
- [X] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged
- [ ] I have added one/more required dependencies
- [X] I have formatted the code using dart format (dartfmt)

**Code/Quality Assurance**
- [X] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

**Mention the dependencies/packages/plugins that have been added (if any)**

### CODE REVIEW

File: pubspec.lock [hunk 1]
```diff
@@ -7,7 +7,7 @@ packages:
       name: async
       url: "https://pub.dartlang.org"
     source: hosted
-    version: "2.5.0-nullsafety.1"
+    version: "2.5.0-nullsafety.3"
```
Comment (werainkhatri) [inline pubspec.lock]: never push pubspec.lock changes unless you have changed pubspec.yaml. Now, retract changes in this specific file and then push to the same branch.

**Review by werainkhatri — CHANGES_REQUESTED** [review]

File: pubspec.lock (hunk 1 above)
Comment (divy-arpit) [reply pubspec.lock]: done!

Comment (werainkhatri) [reply pubspec.lock]: I meant the entire pubspec.lock file, not this specific line.

**Comment by werainkhatri** [comment]
@divy-arpit  Update on this?

**Comment by divy-arpit** [comment]
@werainkhatri that is all done

**Review by werainkhatri — APPROVED** [review]

**Pull request merged**